### PR TITLE
[test] upgrade dependency check version

### PR DESCRIPTION
**Pull request by tejaschumbalkar**
*Issue #, if available:*

## PR Checklist
- [ ] I've prepended PR tag with frameworks/job this applies to : [mxnet, tensorflow, pytorch] | [ei/neuron] | [build] | [test] | [benchmark] | [ec2, ecs, eks, sagemaker]
- [ ] (If applicable) I've documented below the DLC image/dockerfile this relates to
- [ ] (If applicable) I've documented below the tests I've run on the DLC image
- [ ] (If applicable) I've reviewed the licenses of updated and new binaries and their dependencies to make sure all licenses are on the Apache Software Foundation Third Party License Policy Category A or Category B license list.  See [https://www.apache.org/legal/resolved.html](https://www.apache.org/legal/resolved.html).
- [ ] (If applicable) I've scanned the updated and new binaries to make sure they do not have vulnerabilities associated with them.

## Pytest Marker Checklist
- [ ] (If applicable) I have added the marker `@pytest.mark.model("<model-type>")` to the new tests which I have added, to specify the Deep Learning model that is used in the test (use `"N/A"` if the test doesn't use a model)
- [ ] (If applicable) I have added the marker `@pytest.mark.integration("<feature-being-tested>")` to the new tests which I have added, to specify the feature that will be tested
- [ ] (If applicable) I have added the marker `@pytest.mark.multinode(<integer-num-nodes>)` to the new tests which I have added, to specify the number of nodes used on a multi-node test
- [ ] (If applicable) I have added the marker `@pytest.mark.processor(<"cpu"/"gpu"/"eia"/"neuron">)` to the new tests which I have added, if a test is specifically applicable to only one processor type

#### EIA/NEURON Checklist
* When creating a PR:
- [ ] I've modified `src/config/build_config.py` in my PR branch by setting `ENABLE_EI_MODE = True` or `ENABLE_NEURON_MODE = True`
* When PR is reviewed and ready to be merged:
- [ ] I've reverted the code change on the config file mentioned above
#### Benchmark Checklist
* When creating a PR:
- [ ] I've modified `src/config/test_config.py` in my PR branch by setting `ENABLE_BENCHMARK_DEV_MODE = True`
* When PR is reviewed and ready to be merged:
- [ ] I've reverted the code change on the config file mentioned above

## Reviewer Checklist
* For reviewer, before merging, please cross-check:
- [ ] I've verified the code change on the config file mentioned above has already been reverted

*Description:*

We currently see 404 errors while running `dependency-check`. 
```
Downloading 'https://bintray.com/artifact/download/jeremy-long/owasp/dependency-check-5.2.4-release.zip'...
.....
urllib.error.HTTPError: HTTP Error 404: Not Found
```

This is due to the Bintray is [shutting down](https://jfrog.com/blog/into-the-sunset-bintray-jcenter-gocenter-and-chartcenter/): that affects v0.5 versions.
For v0.6, they have moved to GitHub [releases](https://github.com/jeremylong/DependencyCheck/blob/main/RELEASE_NOTES.md#version-602-2020-09-27)

The v0.6 version is not yet available on pypi. Installing using the current [gihub](https://github.com/jhermann/dependency-check-py) repo as a source as the changes for v0.6 release is complete on github but pending on pypi as per the [comment](https://github.com/jhermann/dependency-check-py/issues/7#issuecomment-819891844
) by the owner.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/config/build_config.py [hunk 1]
```diff
@@ -7,7 +7,7 @@
 # Frameworks for which you want to disable both builds and tests
 DISABLE_FRAMEWORK_TESTS = []
 # Disable new builds or build without datetime tag
-DISABLE_DATETIME_TAG = False
+DISABLE_DATETIME_TAG = True
```
Comment (tejaschumbalkar) [inline src/config/build_config.py:10]: To be reverted before PR merge

File: src/config/test_config.py [hunk 2]
```diff
@@ -5,11 +5,11 @@
 # Disable the test codebuild jobs to be run
 
 # It is recommended to set DISABLE_EFA_TESTS to True to disable EFA tests if there is no change to EFA installer version or Frameworks.
-DISABLE_EFA_TESTS = False
+DISABLE_EFA_TESTS = True
```
Comment (tejaschumbalkar) [inline src/config/test_config.py:8]: To be reverted before PR merge

File: test/dlc_tests/sanity/test_pre_release.py [hunk 3]
```diff
@@ -272,21 +272,13 @@ def _run_dependency_check_test(image, ec2_connection, processor):
 @pytest.mark.model("N/A")
 @pytest.mark.canary("Run dependency tests regularly on production images")
 @pytest.mark.parametrize("ec2_instance_type", ["c5.4xlarge"], indirect=True)
```
Comment (tejaschumbalkar) [inline test/dlc_tests/sanity/test_pre_release.py:274]: To be reverted before PR merge

**Review by saimidu — COMMENTED** [review]
Have you tested this by running the modified `testDependencyCheck` script manually, and evaluating the report generated on a container that we expect to pass/fail?

Most of the dependency check result file validation depends on a particular format of output on the report.

Secondly, can we remove the `@pytest.skipif(not canary and not mainline and not nightly)` if we can continue to use GitHub Releases?

**Comment by tejaschumbalkar** [comment]
> LGTM.
> 
> Have you tested this by running the modified `testDependencyCheck` script manually, and evaluating the report generated on a container that we expect to pass/fail?
> 
> Most of the dependency check result file validation depends on a particular format of output on the report.
> 
> Secondly, can we remove the `@pytest.skipif(not canary and not mainline and not nightly)` if we can continue to use GitHub Releases?

Validated the report generated by the new script during local testing.

**Review by saimidu — COMMENTED** [review]
Approved. Please revert the config changes when ready.

**Review by saimidu — APPROVED** [review]

**Pull request merged**